tiledb/cf/xarray_engine/_encoding.py

```diff
@@ -0,0 +1,236 @@
+import numpy as np
+
+import tiledb
+
+# Generic xarray encoding
+_UNLIMITED_DIMS_ENCODING = "unlimited_dims"
+_FILL_VALUE_ENCODING = "_FillValue"
+
+# TileDB specific variable encoding
+_ATTR_NAME_ENCODING = "attr_name"
+_ATTR_FILTERS_ENCODING = "filters"
+_TILE_SIZES_ENCODING = "tiles"
+_MAX_SHAPE_ENCODING = "max_shape"
+_DIM_DTYPE_ENCODING = "dim_dtype"
+
+
+class TileDBVariableEncoder:
+    valid_encoding_keys = {
+        _ATTR_FILTERS_ENCODING,
+        _ATTR_NAME_ENCODING,
+        _DIM_DTYPE_ENCODING,
+        _FILL_VALUE_ENCODING,
+        _MAX_SHAPE_ENCODING,
+        _TILE_SIZES_ENCODING,
+    }
+
+    def __init__(self, name, variable, encoding, unlimited_dims, ctx):
+        """Creates a class for encoding xarray variables.
+
+        Parameters:
+        -----------
+        name: Name of the variable.
+        variable: Xarray variable to encode.
+        encoding: Dictionary of TileDB encoding keywords.
+        unlimited_dims: Unlimited dimensions. Only used if max_shape is not provided
+            in the encoding.
+        ctx: Context object for TileDB operations.
+        """
+        # Set initial class properties.
+        self._ctx = ctx
+        self._name = name
+        self._variable = variable
+        self._encoding = dict()
+
+        # Check the input encoding data is valid.
+        for key in encoding:
+            if key not in self.valid_encoding_keys:
+                raise KeyError(
+                    "Encoding error on variable '{self._name}'. Invalid encoding key "
+                    f"``{key}``."
+                )
+
+        # Initialize encoding values.
+        try:
+            # Set attribute encodings: attr_name, attr_filters, and attr_fill.
+            self.attr_name = encoding.get(
+                _ATTR_NAME_ENCODING,
+                f"{self._name}_" if self._name in variable.dims else self._name,
```

Comment:
Coordinates in xarray generally have matching variable and dimension names. The convention here is just adding an `_` after the variable name. Is there a better choice?